contracts/pallet-ibc/src/lib.rs

```diff
@@ -697,6 +717,57 @@ pub mod pallet {
 				return Err(Error::<T>::InvalidTimestamp.into())
 			}
 
+			let mut ctx = Context::<T>::default();
+			let channel_end = ctx
+				.channel_end(&(PortId::transfer(), source_channel))
+				.map_err(|_| Error::<T>::ChannelNotFound)?;
+
+			let destination_channel = channel_end
+				.counterparty()
+				.channel_id
+				.ok_or_else(|| Error::<T>::ChannelNotFound)?;
+
+			let whitelisted = <WhitelistChannelIds<T>>::contains_key((
+				source_channel.sequence(),
+				destination_channel.sequence(),
+			));
+
+			if !whitelisted {
+				let percent = ServiceCharge::<T>::get().unwrap_or(T::ServiceCharge::get());
+				// Now we proceed to send the service fee from the receiver's account to the pallet
+				// FeeAccount
+				let fee_account = T::FeeAccount::get();
+
+				//TODO check the validity of the statment fee_coin.
```

Comment:
need to add same logic as on destination. so we need to check for flat fee

Reply:
Done.